File: corelibrary/src/main/java/pers/cxd/corelibrary/util/ViewAnimUtil.java
package pers.cxd.corelibrary.util;

import android.view.View;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.view.animation.TranslateAnimation;

/**
 * Miscellaneous {@link Animation} utility methods.
 *
 * @author pslilysm
 * @since 1.0.0
 */
public class ViewAnimUtil {

    private static final int DEFAULT_ANIM_DURATION = 250;

    public static void showViewWithAlpha(View v) {
        if (v.getVisibility() != View.VISIBLE) {
            AlphaAnimation alphaAnim = new AlphaAnimation(0, 1);
            alphaAnim.setDuration(DEFAULT_ANIM_DURATION);
            alphaAnim.setFillAfter(true);
            v.startAnimation(alphaAnim);
        }
    }

    public static void hideViewWithAlpha(View v) {
        if (v.getVisibility() == View.VISIBLE) {
            AlphaAnimation alphaAnim = new AlphaAnimation(1, 0);
            alphaAnim.setDuration(DEFAULT_ANIM_DURATION);
//            alphaAnim.setFillAfter(true);
            v.startAnimation(alphaAnim);
        }
    }

    public static void showViewWithTranslateFromBottomToTop(View v) {
        if (v.getVisibility() != View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0,
                    Animation.RELATIVE_TO_SELF, 1, Animation.RELATIVE_TO_SELF, 0);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            transAnim.setFillAfter(true);
            v.startAnimation(transAnim);
        }
    }

    public static void hideViewWithTranslateFromTopToBottom(View v) {
        if (v.getVisibility() == View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0,
                    Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 1);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            v.startAnimation(transAnim);
        }
    }

    public static void translateFromLeftToRight(View v) {
        TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 1,
                Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
        transAnim.setDuration(DEFAULT_ANIM_DURATION);
        transAnim.setFillAfter(true);
        v.startAnimation(transAnim);
    }

    public static void translateFromRightToLeft(View v) {
        TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 1, Animation.RELATIVE_TO_SELF, 0,
                Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
        transAnim.setDuration(DEFAULT_ANIM_DURATION);
        transAnim.setFillAfter(true);
        v.startAnimation(transAnim);
    }

    public static void hideViewWithTranslateFromRightToLeft(View v) {
        if (v.getVisibility() == View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, -1,
                    Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            v.startAnimation(transAnim);
        }
    }

    public static void showViewWithTranslateFromRightToLeft(View v) {
        if (v.getVisibility() != View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 1, Animation.RELATIVE_TO_SELF, 0,
                    Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            v.startAnimation(transAnim);
        }
    }

    public static void hideViewWithTranslateFromLeftToRight(View v) {
        if (v.getVisibility() == View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 1,
                    Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            v.startAnimation(transAnim);
        }
    }

    public static void showViewWithTranslateFromLeftToRight(View v) {
        if (v.getVisibility() != View.VISIBLE) {
            TranslateAnimation transAnim = new TranslateAnimation(Animation.RELATIVE_TO_SELF, -1, Animation.RELATIVE_TO_SELF, 0,
                    Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0);
            transAnim.setDuration(DEFAULT_ANIM_DURATION);
            v.startAnimation(transAnim);
        }
    }

}

File: corelibrary/src/main/java/pers/cxd/corelibrary/util/ViewAnimUtil.kt
package pers.cxd.corelibrary.util

import android.view.View
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.view.animation.TranslateAnimation

/**
 * Miscellaneous [Animation] utility methods.
 *
 * @author pslilysm
 * @since 1.0.0
 */
object ViewAnimUtil {
    private const val DEFAULT_ANIM_DURATION = 250

    @kotlin.jvm.JvmStatic
    fun showViewWithAlpha(v: View) {
        if (v.visibility != View.VISIBLE) {
            val alphaAnim = AlphaAnimation(0f, 1f)
            alphaAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            alphaAnim.fillAfter = true
            v.startAnimation(alphaAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun hideViewWithAlpha(v: View) {
        if (v.visibility == View.VISIBLE) {
            val alphaAnim = AlphaAnimation(1f, 0f)
            alphaAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            //            alphaAnim.setFillAfter(true);
            v.startAnimation(alphaAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun showViewWithTranslateFromBottomToTop(v: View) {
        if (v.visibility != View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
                Animation.RELATIVE_TO_SELF, 1f, Animation.RELATIVE_TO_SELF, 0f
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            transAnim.fillAfter = true
            v.startAnimation(transAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun hideViewWithTranslateFromTopToBottom(v: View) {
        if (v.visibility == View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 1f,
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            v.startAnimation(transAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun translateFromLeftToRight(v: View) {
        val transAnim = TranslateAnimation(
            Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 1f,
            Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
        )
        transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
        transAnim.fillAfter = true
        v.startAnimation(transAnim)
    }

    @kotlin.jvm.JvmStatic
    fun translateFromRightToLeft(v: View) {
        val transAnim = TranslateAnimation(
            Animation.RELATIVE_TO_SELF, 1f, Animation.RELATIVE_TO_SELF, 0f,
            Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
        )
        transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
        transAnim.fillAfter = true
        v.startAnimation(transAnim)
    }

    @kotlin.jvm.JvmStatic
    fun hideViewWithTranslateFromRightToLeft(v: View) {
        if (v.visibility == View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, -1f,
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            v.startAnimation(transAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun showViewWithTranslateFromRightToLeft(v: View) {
        if (v.visibility != View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, 1f, Animation.RELATIVE_TO_SELF, 0f,
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            v.startAnimation(transAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun hideViewWithTranslateFromLeftToRight(v: View) {
        if (v.visibility == View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 1f,
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            v.startAnimation(transAnim)
        }
    }

    @kotlin.jvm.JvmStatic
    fun showViewWithTranslateFromLeftToRight(v: View) {
        if (v.visibility != View.VISIBLE) {
            val transAnim = TranslateAnimation(
                Animation.RELATIVE_TO_SELF, -1f, Animation.RELATIVE_TO_SELF, 0f,
                Animation.RELATIVE_TO_SELF, 0f, Animation.RELATIVE_TO_SELF, 0f,
            )
            transAnim.duration = DEFAULT_ANIM_DURATION.toLong()
            v.startAnimation(transAnim)
        }
    }
}